packages/legend-query/src/stores/QueryBuilderDateTimeHelper.ts

```diff
@@ -0,0 +1,597 @@
+        value: QUERY_BUILDER_CUSTOM_DATE_START_DATE_OPTION.START_OF_YEAR,
+      };
+    default:
+      return { label: '', value: '' };
+  }
+};
+
+export const generateDateOption = (
+  valueSpecification: SimpleFunctionExpression | PrimitiveInstanceValue,
+  graph: PureModel,
+): CustomDateOption => {
+  if (valueSpecification instanceof SimpleFunctionExpression) {
+    switch (valueSpecification.functionName) {
+      case SUPPORTED_FUNCTIONS.TODAY:
+        return {
+          label: QUERY_BUILDER_DATE_OPTION.TODAY,
+          value: QUERY_BUILDER_DATE_OPTION.TODAY,
+        };
+      case SUPPORTED_FUNCTIONS.NOW:
+        return {
+          label: QUERY_BUILDER_DATE_OPTION.NOW,
+          value: QUERY_BUILDER_DATE_OPTION.NOW,
+        };
+      case SUPPORTED_FUNCTIONS.FIRST_DAY_OF_YEAR:
+        return {
+          label: QUERY_BUILDER_CUSTOM_DATE_START_DATE_OPTION.START_OF_YEAR,
+          value: QUERY_BUILDER_DATE_OPTION.START_DAY_OF_DATE,
+          unit: QUERY_BUILDER_DATE_DURATION_UNIT.YEAR,
+        };
+      case SUPPORTED_FUNCTIONS.FIRST_DAY_OF_QUARTER:
+        return {
+          label: QUERY_BUILDER_CUSTOM_DATE_START_DATE_OPTION.START_OF_QUARTER,
+          value: QUERY_BUILDER_DATE_OPTION.START_DAY_OF_DATE,
+          unit: QUERY_BUILDER_DATE_DURATION_UNIT.QUARTER,
+        };
+      case SUPPORTED_FUNCTIONS.FIRST_DAY_OF_MONTH:
+        return {
+          label: QUERY_BUILDER_CUSTOM_DATE_START_DATE_OPTION.START_OF_MONTH,
+          value: QUERY_BUILDER_DATE_OPTION.START_DAY_OF_DATE,
+          unit: QUERY_BUILDER_DATE_DURATION_UNIT.MONTH,
+        };
+      case SUPPORTED_FUNCTIONS.FIRST_DAY_OF_WEEK:
+        return {
+          label: QUERY_BUILDER_CUSTOM_DATE_START_DATE_OPTION.START_OF_WEEK,
+          value: QUERY_BUILDER_DATE_OPTION.START_DAY_OF_DATE,
+          unit: QUERY_BUILDER_DATE_DURATION_UNIT.WEEK,
+        };
+      case SUPPORTED_FUNCTIONS.PREVIOUS_DAY_OF_WEEK:
+        return {
+          label: `Previous ${
+            (valueSpecification.parametersValues[0] as EnumValueInstanceValue)
+              .values[0]?.value.name
+          }`,
+          value: QUERY_BUILDER_DATE_OPTION.PREVIOUS_DAY_OF_WEEK,
+          unit: (
+            valueSpecification.parametersValues[0] as EnumValueInstanceValue
+          ).values[0]?.value.name,
+        };
+      case SUPPORTED_FUNCTIONS.ADJUST:
+        return {
+          ...generateDateOptionForAdjustFunction(valueSpecification, graph),
+        };
+      default:
+        return { label: '', value: '' };
+    }
+  } else {
+    return {
+      label: valueSpecification.values[0] as string,
+      value:
+        valueSpecification.genericType.value.rawType.path ===
+        PRIMITIVE_TYPE.DATETIME
+          ? QUERY_BUILDER_DATE_OPTION.ABSOLUTE_TIME
+          : QUERY_BUILDER_DATE_OPTION.ABSOLUTE_DATE,
+    };
+  }
+};
+
+export const chunkArray = (arr: unknown[], stepSize: number): unknown[][] => {
```

Comment:
should probably be in `legend-shared`

Reply:
This should be removed. I moved this function into BaseRadioGroup.tsx, and renamed it to `transformToMatrix`.